### PR TITLE
Update pu/pu to support deployment run command

### PR DESCRIPTION
**Pull request by glena**
# Description

Add new deployment run command

Fixes https://github.com/pulumi/pulumi-service/issues/20500

## Checklist

- [ ] I have run `make tidy` to update any new dependencies
- [ ] I have run `make lint` to verify my code passes the lint check
  - [ ] I have formatted my code using `gofumpt`

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/cmd/pulumi/deployment.go
Comment (komalali) [inline pkg/cmd/pulumi/deployment.go:150]: What does the json output look like?

File: pkg/cmd/pulumi/deployment.go
Comment (komalali) [inline pkg/cmd/pulumi/deployment.go:142]: Does this work for deployments?

**Review by komalali — APPROVED** [review]

File: pkg/cmd/pulumi/deployment.go
Comment (glena) [reply pkg/cmd/pulumi/deployment.go:150]: Actually it doesnt work, I will remove it

**Comment by MMartyn** [comment]
Just curious if it is intentional to not have a flag for agent pool. Thanks.

File: pkg/cmd/pulumi/deployment.go
Comment (glena) [reply pkg/cmd/pulumi/deployment.go:138]: yes, but it is already this way in `pulumi up` and I left as a string bar for consistency (if people is already used to use it that way it would be weird). Happy to change it to BoolVar tho
 https://github.com/pulumi/pulumi/blob/f69f1d2ee3cdc98c56aaad0094c5b474a1a062df/pkg/cmd/pulumi/up.go#L627-L629

File: pkg/cmd/pulumi/util_remote.go
Comment (glena) [reply pkg/cmd/pulumi/util_remote.go:202]: no, those are parameters sent when the deployment job is created. This is currently supported by the automation api and now we are exposing it in a command

https://github.com/pulumi/pulumi/blob/f69f1d2ee3cdc98c56aaad0094c5b474a1a062df/pkg/cmd/pulumi/util_remote.go#L204-L207

File: pkg/cmd/pulumi/util_remote.go
Comment (glena) [reply pkg/cmd/pulumi/util_remote.go:179]: makes sense, done

File: pkg/cmd/pulumi/util_remote.go
Comment (tgummerer) [reply pkg/cmd/pulumi/util_remote.go:202]:
```suggestion
		"Git commit hash of the commit to deploy; "+
```

Great, I don't think we need to specify `HEAD will be in detached mode` then, since this should not be relevant to the user.  They just need to know that they'll deploy this particular commit hash.  In fact, I think we should also remove this text in `util_remote.go`, though that would be in a separate PR.

File: pkg/backend/httpstate/backend.go
Comment (tgummerer) [inline pkg/backend/httpstate/backend.go:176]:
```suggestion
		opts display.Options, deploymentInitiator string, suppressStreamLogs bool) error
```

File: pkg/backend/httpstate/backend.go
Comment (tgummerer) [inline pkg/backend/httpstate/backend.go:1920]: Since we inverted the flag, this should now also be inverted.

File: pkg/cmd/pulumi/deployment.go
Comment (glena) [reply pkg/cmd/pulumi/deployment.go:138]: ahh I missed the usage of NoOptDefVal, yes I will change to BoolVar then!

File: pkg/cmd/pulumi/util_remote.go
Comment (glena) [reply pkg/cmd/pulumi/util_remote.go:202]: well `HEAD will be in detached mode` is still true for the deployment, it will affect the program running there (the program running in the deployment job)

Comment (tgummerer) [reply pkg/cmd/pulumi/util_remote.go:202]: fair enough!

**Comment by glena** [comment]
> Just curious if it is intentional to not have a flag for agent pool. Thanks.

@MMartyn I was going to add support later, but I just added it here so I dont forget

File: pkg/backend/httpstate/backend.go
Comment (glena) [reply pkg/backend/httpstate/backend.go]: yep

File: pkg/backend/httpstate/backend.go
Comment (glena) [reply pkg/backend/httpstate/backend.go:176]: done

File: sdk/go/common/apitype/deployments.go
Comment (glena) [inline sdk/go/common/apitype/deployments.go:77]: @tgummerer I added this piece to handle pool id mashalling (see util_remote usage)

File: pkg/cmd/pulumi/util_remote.go
Comment (glena) [inline pkg/cmd/pulumi/util_remote.go:499]: @tgummerer added this piece to handle pool id

File: pkg/backend/httpstate/mock.go
Comment (tgummerer) [inline pkg/backend/httpstate/mock.go:44]: We should probably rename this as well to `suppressStreamLogs`.  Sorry I forgot pointing this out earlier.

File: sdk/go/common/apitype/deployments.go
Comment (glena) [inline sdk/go/common/apitype/deployments.go:83]: actually this is not needed due the omitempty, I am cleaning it up

**Review by tgummerer — APPROVED** [review]
One small rename nit, but other than that this looks good to me!

File: pkg/backend/httpstate/mock.go
Comment (glena) [reply pkg/backend/httpstate/mock.go:44]: ah damn! good catch

**Comment by justinvp** [comment]
Sorry, need to hold off on merging this for a little while. I'll add it back to the queue in a little bit after some higher priority fixes for CI have merged.

**Comment by glena** [comment]
@justinvp np, it can wait till monday

**Pull request merged**
